stats.go

```diff
@@ -34,3 +34,109 @@ func Median(input []float64) (output float64) {
 
 	return output
 }
+
+// Max returns the maximum value of the sample
+func Max(input []float64) float64 {
+	if len(input) == 0 {
+		return math.NaN()
+	}
+	max := input[0]
+	for i := 1; i < len(input); i++ {
+		if input[i] > max {
+			max = input[i]
+		}
+	}
+	return max
+}
+
+// Min returns the minimum value of the sample
+func Min(input []float64) float64 {
+	if len(input) == 0 {
+		return math.NaN()
+	}
+	min := input[0]
+	for i := 1; i < len(input); i++ {
+		if input[i] < min {
+			min = input[i]
+		}
+	}
+	return min
+}
+
+// Range returns the difference between the largest and smallest values
+func Range(input []float64) float64 {
+	if len(input) < 2 {
+		return math.NaN()
+	}
+	return Max(input) - Min(input)
+}
+
+// sumOfSquaredDifferences returns the sum of the squared differences of each observation from the mean
+func sumOfSquaredDifferences(input []float64) float64 {
+	if len(input) < 2 {
+		return math.NaN()
+	}
+	mean := Mean(input)
+	ssd := 0.0
+	for _, o := range input {
+		ssd += math.Pow(o-mean, 2.0)
+	}
+	return ssd
+}
+
+// StandardDeviationSample returns the standard deviation of the sample
+func StandardDeviationSample(input []float64) float64 {
```

Comment:
Why adding `Sample` at the end? I think `StdDev` is enough and generally understood.

Reply:
Because the formula is different if you calculate the StdDev of a Sample or of a Population

Reply:
I wanted also to implement StandardDeviationPopulation

Reply:
I would assume that if nothing else is specified the `StdDev` is the sampled versio, and I think that 99% of people using this function are also expecting it.
If later you want to add the Population version you can always add this keyword to the end of the name.

Reply:
I agree with you. Sampled version is the most common use.